shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -170,6 +170,8 @@ - (void)setViewController:(FlutterViewController*)viewController {
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
 
+  [_flutterViewControllerWillDeallocObserver release];
```

Comment:
@xster I am not sure if this the appropriate fix.
I ran into an issue that calling setViewController multiple time will register multiple observer for flutter vc. This happens in one of my test where i do [engine setViewController:nil] at the end which causes notifyViewControllerDeallocated to be called twice.

I think it make more sense to release the previous one in the case of setting multiple times

Reply:
@gaaclarke, I have a feeling we had this discussion but didn't do it for some reason. You might have a better recollection of the reasoning.